src/crimson/osd/object_context_loader.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <seastar/core/future.hh>
+#include "crimson/common/errorator.h"
+#include "crimson/osd/object_context.h"
+#include "crimson/osd/shard_services.h"
+#include "crimson/osd/pg_backend.h"
+
+namespace crimson::osd {
+class ObjectContextLoader {
+public:
+  using obc_accessing_list_t = boost::intrusive::list<
+    ObjectContext,
+    ObjectContext::obc_accessing_option_t>;
+
+  ObjectContextLoader(
+    ShardServices& _shard_services,
+    PGBackend* _backend,
+    obc_accessing_list_t& _set_accessing)
+    : shard_services{_shard_services},
+      backend{_backend},
+      obc_set_accessing{_set_accessing}
```

Comment:
obc_set_accessing should be owned by ObjectContextLoader.  The only usage not moved over is in PG::on_change, which should be changed to something like ObjectContextLoader::notify_on_change().

Reply:
I ended up using `ObjectContextLoader::obc_set_accessing` in PG::on_change. (9a42cfde19dfc314fea36514e77fb621e66edc42)